### PR TITLE
Math updates

### PR DESCRIPTION
**Pull request by dgarske**
* Adds `TFM_MIPS`.
* Adds check for `mp_read_radix` to return failure for invalid character.
* Adds checking for `s_mp_sub` response code.
* Improved `mp_init_copy` error handling.
* Improved init to use size where applicable.
* Fixes for PPC32.
* Minor cleanups.

### CODE REVIEW

**Comment by dgarske** [comment]
Note: wolfCrypt test on MIPS (PIC32MZ) with TFM_MIPS option failed on an RSA sig test `RSA      test failed! error = -5358`. So it looks like TFM_MIPS optimization isn't working correctly (tests pass without TFM_MIPS). I'll pursue this after the PR #357 work.

**Comment by dgarske** [comment]
@toddouska : Fixed `TFM_MIPS`. Tested on PIC32MZ2048EFM144 with XC32 v1.44. This PR is ready.

File: wolfcrypt/src/asm.c [hunk 1]
```diff
@@ -445,7 +445,7 @@ __asm__(                           \
    "paddq %%mm5,%%mm3    \n\t" \
    "movd %%mm3,28(%0)    \n\t" \
    "psrlq $32, %%mm3     \n\t" \
-:"=r"(_c) : "0"(_c), "r"(tmpm) );
```
Comment (JacobBarthelmeh) [inline wolfcrypt/src/asm.c]: Why was this reverted back to "g"? There is a comment below of why it had been changed to "r".

File: wolfcrypt/src/asm.c [hunk 2]
```diff
@@ -552,46 +552,46 @@ __asm__(                             \
...
-:"=r"(cy),"=m"(_c[0]):"0"(cy),"r"(mu),"r"(tmpm[0]),"1"(_c[0]):"16", "17", "18","cc"); ++tmpm;
-
-#define PROPCARRY                    \
-__asm__(                                 \
-   " ldx      16,0,%1       \n\t"    \
-   " addc     16,16,%0      \n\t"    \
-   " sdx      16,0,%1       \n\t"    \
-   " xor      %0,%0,%0      \n\t"    \
-   " addze    %0,%0         \n\t"    \
-:"=r"(cy),"=m"(_c[0]):"0"(cy),"1"(_c[0]):"16","cc");
```
Comment (JacobBarthelmeh) [inline wolfcrypt/src/asm.c]: Was this change tested with 32bit system?

File: wolfcrypt/src/asm.c [hunk 3]
```diff
@@ -751,18 +783,16 @@ __asm__(                                                     \
...
+__asm__(                                                  \
      "movq  %3,%%rax     \n\t"                            \
      "mulq  %4           \n\t"                            \
      "movq  %%rax,%0     \n\t"                            \
      "movq  %%rdx,%1     \n\t"                            \
      "xorq  %2,%2        \n\t"                            \
      :"=r"(sc0), "=r"(sc1), "=r"(sc2): "g"(i), "g"(j) :"%rax","%rdx","cc");
 
-/* TAO removed sc0,1,2 as input to remove warning so %6,%7 become %3,%4 */
-
```
Comment (JacobBarthelmeh) [inline wolfcrypt/src/asm.c]: why was this comment removed?

File: wolfcrypt/src/asm.c (hunk 1 above)
Comment (dgarske) [reply wolfcrypt/src/asm.c]: I will revert this.

File: wolfcrypt/src/asm.c (hunk 2 above)
Comment (dgarske) [reply wolfcrypt/src/asm.c]: I will re-test on PPC 32-bit.

File: wolfcrypt/src/asm.c (hunk 3 above)
Comment (dgarske) [reply wolfcrypt/src/asm.c]: This change is default in the stock fast math code as well, so there is no customization here anymore.

**Comment by JacobBarthelmeh** [comment]
Thanks

**Pull request merged**